readers/mocks/messages.go

```diff
@@ -29,7 +29,45 @@ func NewMessageRepository(messages map[string][]mainflux.Message) readers.Messag
 	}
 }
 
-func (repo *messageRepositoryMock) ReadAll(chanID string, offset, limit uint64, query map[string]string) (readers.MessagesPage, error) {
+func (repo *messageRepositoryMock) ReadAllByThingID(publisher string, offset, limit uint64, query map[string]string) (readers.MessagesPage, error) {
```

Comment:
You need `chanID` as well, as we want to reall all messages for this particular channel, but then filter them by `thingID`.

This is why I proposed vriadic paramter `thingID`, but maybe this would become confusing as we add more and more filters.

Reply:
Currently, we have this filter on the `query` field. You can provide `publisher field on `query` and then you get messages from the specific channel with given `publisher`. Here an example from MongoDB reader.

https://github.com/mainflux/mainflux/blob/97327ab05fc4426569843c36077f581029291741/readers/mongodb/messages.go#L131

Reply:
@1995parham any update on this? All is missing is Postgres reader update, to align it to other readers that can filter by `thingID`.

Reply:
@drasko I think we can have another PR for adding this filtering feature to Postgres because of current changes in this PR is big. Do you agree?